tests/testsuite/cargo_add/symlink.rs

```diff
@@ -0,0 +1,51 @@
+use cargo_test_support::prelude::*;
+use cargo_test_support::project;
+use std::fs;
+
+#[cargo_test]
+fn preserve_symlink_to_cargo_toml() {
+    if !cargo_test_support::symlink_supported() {
+        return;
+    }
+
+    let project = project().file("src/lib.rs", "").build();
+
+    let target_dir = project.root().join("target_dir");
+    fs::create_dir_all(&target_dir).unwrap();
+
+    // Create the target Cargo.toml
+    fs::copy(
+        project.root().join("Cargo.toml"),
+        target_dir.join("Cargo.toml"),
+    )
+    .unwrap();
+
+    // Remove the original and create a symlink
+    fs::remove_file(project.root().join("Cargo.toml")).unwrap();
+
+    #[cfg(unix)]
+    {
+        use std::os::unix::fs::symlink;
+        symlink(
+            target_dir.join("Cargo.toml"),
+            project.root().join("Cargo.toml"),
+        )
+        .unwrap();
+    }
+
+    #[cfg(windows)]
+    {
+        use std::os::windows::fs::symlink_file;
+        symlink_file(
+            target_dir.join("Cargo.toml"),
+            project.root().join("Cargo.toml"),
+        )
+        .unwrap();
+    }
+
+    // Add a dependency
+    project.cargo("add rand").run();
```

Comment:
This is pulling an exteranl dependency from crates.io. We avoid that in cargo's testsuite and instead host a [local registry fixure](https://doc.rust-lang.org/1.85.0/nightly-rustc/cargo_test_support/registry/index.html), like

* https://github.com/rust-lang/cargo/blob/ab1463d632528e39daf35f263e10c14cbe590ce8/tests/testsuite/rename_deps.rs?plain=1#L12
* https://github.com/rust-lang/cargo/blob/ab1463d632528e39daf35f263e10c14cbe590ce8/tests/testsuite/cargo_add/features/mod.rs?plain=1#L10-L16